integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/KafkaWriterVisitor.java

```diff
@@ -0,0 +1,36 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.PlanUtils;
+import io.openlineage.spark.agent.util.ScalaConversionUtils;
+import java.util.Collections;
+import java.util.List;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand;
+import org.apache.spark.sql.kafka010.KafkaSourceProvider;
+
+public class KafkaWriterVisitor
+    extends QueryPlanVisitor<SaveIntoDataSourceCommand, OpenLineage.Dataset> {
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan x) {
+    return x instanceof SaveIntoDataSourceCommand
+        && ((SaveIntoDataSourceCommand) x).dataSource() instanceof KafkaSourceProvider;
+  }
+
+  @Override
+  public List<OpenLineage.Dataset> apply(LogicalPlan x) {
+    SaveIntoDataSourceCommand saveIntoDataSourceCommand = (SaveIntoDataSourceCommand) x;
+
+    String servers =
+        ScalaConversionUtils.asJavaOptional(
+                saveIntoDataSourceCommand.options().get("kafka.bootstrap.servers"))
+            .orElse("");
+    String namespace = "kafka://" + servers.split(",")[0];
+    String topic =
+        ScalaConversionUtils.asJavaOptional(saveIntoDataSourceCommand.options().get("topic"))
```

Comment:
There are two options how we can write to kafka.
1. By specifying topic in options
2. By specifying topic in column (we can't get that info) <- this case is not covered, cause we can get only column names, we can't get value, and it can be different per each row

```
First option
df.selectExpr("CAST(key AS STRING)", "CAST(value AS STRING)")
  .write
  .format("kafka")
  .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
  .option("topic", "topic1")
  .save()

Second option 
// Write key-value data from a DataFrame to Kafka using a topic specified in the data
df.selectExpr("topic", "CAST(key AS STRING)", "CAST(value AS STRING)")
  .write
  .format("kafka")
  .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
  .save()

```